### ISSUE TITLE
valid `RET507 Unnecessary `elif` after `continue` statement` ?

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#2216]
I get `RET507` Unnecessary `elif` after `continue` statement for:

```
def something(stdout):
    for line in stdout:
        if line.startswith('foo'):
            pass
        elif not line.strip():
            continue
        elif line.startswith('bar'):
            pass
        elif line.startswith('baz'):
            pass
```

I consider this code to be okay, that we should not transform the `elif` to a new `if`:

Btw. if the code is not inside of a function the `RET507` is not detected.

**Comment by spaceone** [comment]
same for `RET508` where `break` instead of `continue` is used.

**Comment by charliermarsh** [comment]
This is consistent with the upstream rule. I agree it's somewhat confusing, but I believe it's "correct" given the rule.
